app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -46,13 +60,31 @@ class WallpaperManager(
             wallpaperContainer.setBackgroundColor(context.getColorFromAttr(DEFAULT_RESOURCE))
             logger.info("Wallpaper update to default background")
         } else {
-            logger.info("Wallpaper update to ${newWallpaper.name}")
-            val bitmap = loadWallpaperFromAssets(newWallpaper, context)
-            wallpaperContainer.background = BitmapDrawable(context.resources, bitmap)
+            val bitmap = if (newWallpaper.themeCollection.origin == WallpaperOrigin.Local) {
+                loadWallpaperFromAssets(newWallpaper, context)
+            } else {
+                loadWallpaperFromDisk(context, newWallpaper)
+            }
+            if (bitmap == null) {
+                logger.error("Could not load wallpaper bitmap. Resetting to default")
```

Comment:
Similar to https://github.com/mozilla-mobile/fenix/pull/23505#discussion_r796798226

Reply:
Added a crash report!